### PR TITLE
Fix dollar quotes for multiline support + spaces and comments afterward

### PR DESCRIPTION
**Pull request by dgehlich**
Hi, regarding https://github.com/tkopets/sublime-postgresql-syntax/pull/2
this seems the fix for booth cases:
* spaces and comments after function body
* multiline double dollar quotes

```
CREATE OR REPLACE FUNCTION dg_quote_1()
RETURNS VOID AS $$ -- some comment with space
  BEGIN
    SELECT this_code_but_no_comment;
    SELECT $q1$some inner quote$q1$;
END $$ LANGUAGE plpgsql STABLE;


COMMENT ON FUNCTION dg_quote_1() IS
-- works now
$DOC$multiline
documentation$DOC$

$DOC2$singlelinedocumentation$DOC2$
;
```

grafik

### CODE REVIEW

**Comment by tkopets** [comment]
Thanks, Dominik!

File: PostgreSQL.YAML-tmLanguage [hunk 1]
```diff
@@ -84,12 +84,12 @@ repository:
...
     patterns:
     - comment: "Assume multiline dollar quote is SQL body!
-        Start if double dollar quote is followed by no dollar quote till line ending.
+        Start if double dollar quote is followed by comment (-- or /**/) or linebreak (\n or \r).
         See match for dollar quotes as string: string.unquoted.dollar.pgsql.
         This could easily support other PL languages like PHP and Ruby -- see PHP heredoc as an example."
       # name: meta.dollar-quote.pgsql
       contentName: meta.dollar-quote.pgsql
-      begin: (\$[\w_]*\$)(?=[^\$]*$)
+      begin: (\$[\w_]*\$)(?:\s*)(?=[-\/\n\r]+)
```
Comment (tkopets) [inline PostgreSQL.YAML-tmLanguage]: Minor...
Maybe we push that whitespace match `(?:\s*)` into the lookahead itself to make it a bit simpler?
Something like this
`(\$[\w_]*\$)(?=\s*[-\/\n\r]+)`

Comment (tkopets) [reply PostgreSQL.YAML-tmLanguage]: Whatever you think is best in this case.
Merging this PR regardless of your decision on this.

**Comment by tkopets** [comment]
Also thank you for providing the screenshots on all your PR's for easy evaluation.

**Comment by dgehlich** [comment]
fixed as u suggested

**Comment by dgehlich** [comment]
conflicts resolved

**Comment by dgehlich** [comment]
nice. thanks for the good team work

**Comment by tkopets** [comment]
Thanks for your efforts!
The [release](https://github.com/tkopets/sublime-postgresql-syntax/releases/tag/1.0.6) is on the way and should hit Package Control soonish.

**Comment by sbuttgereit** [comment]
I just got the package from Package Control.  The application I'm working with has, including overloaded functions, ~2000 or so functions defined; including functions that were written as far back as the the old PostgreSQL 7.4 days, long before dollar quoting existed in PostgreSQL, to functions written as recently as, well, yesterday.

This change looks pretty good.  I haven't looked at all the functions, but glancing over a cross section of 100 or so would indicate that this has fixed for the cases I was having trouble with and I haven't seen any new issues crop up.  Old style functions, written before dollar quoting, have function bodies that look like all quoted text... but I think that's a feature more than a bug: definitely nothing I'd recommend you try to account for.

Thanks so much for taking a look at this and jumping on the issue.  Sorry I couldn't have helped more.

**Comment by dgehlich** [comment]
The bug report and fix confirmation itself was helpfull!
Good to here everything is fine now.

Our devs and me are working with this highlighter on a big codebase, too. Thus I'll get feedback and a wishlist.

I'll keep you up to date with pull requests. :D


Btw. some flaw with linebreaks:
```
CREATE OR REPLACE FUNCTION func() RETURNS void AS $$
BEGIN
  EXECUTE $a$ code a $a$ || concat || $b$ code b $b$;
  
  -- vs
  EXECUTE
      $a$
        code a
      $a$
      
      || concat ||
      
      $b$
        code b
      $b$
  ;
END $$ LANGUAGE plpgsql;
```
grafik

I think there is no simple way to fix this.
Except re code the highlighter just for double dollar quotes. The idea is: get $$ as string by keywords (ie EXECUTE, IS or not AS). I dont want to do this, cause we have no need.

**Comment by dgehlich** [comment]
I will call it a feature. `¯\_(ツ)_/¯`

**Pull request merged**